### PR TITLE
Mostleast accurate team

### LINKED ISSUE #26: Most/Least Accurate Team

### PR DESCRIPTION
**Pull request by topher-nullset**
## Describe your changes in detail
Added most and least accurate teams methods with a helper method to calculate team accuracies.
added the tests for these methods

## Checklist before requesting a review
- [ x ] I have performed a self-review of my code
- [ x ] I have added thorough tests
- [ x ] I AM MERGING WITH Westonio/futbol

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -142,13 +142,31 @@ def worst_coach
...
-
+    team_accuracies = calculate_team_accuracies
+    team_accuracies.max_by { |team, _| team_accuracies[team] }[0]
   end
 
   def least_accurate_team
+    team_accuracies = calculate_team_accuracies
+    team_accuracies.min_by { |team, _| team_accuracies[team] }[0]
+  end
+
```
Comment (pcbennett108) [inline lib/stat_tracker.rb:189]: Nice use of helper method.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -142,13 +142,31 @@ def worst_coach
...
+    end
 
+    team_accuracies = {}
+    team_goals.each do |team_id, (goals, shots)|
+      accuracy = (goals.to_f / shots.to_f) * 100
+      team_name = team_name_by_id(team_id)
+      team_accuracies[team_name] = accuracy.round(2)
+    end
+
+    team_accuracies
```
Comment (westonio) [inline lib/stat_tracker.rb]: Looks like these tests are failing. Do you mind taking a look to make sure everything is operating as intended?

Comment (westonio) [reply lib/stat_tracker.rb]: Woops, sorry! These most_accurate and least_accurate team methods also need to take an argument of the season ID

Comment (pcbennett108) [reply lib/stat_tracker.rb]: Be aware that the 'season' field in the CSV is a string of 8 chars. i.e. "20122013" ect.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (fadwil) [reply lib/stat_tracker.rb:189]: I agree this makes the following code look clean

File: lib/stat_tracker.rb (hunk 2 above)
Comment (fadwil) [reply lib/stat_tracker.rb]: Good catch Weston!

Comment (westonio) [reply lib/stat_tracker.rb]: @topher-nullset Thanks for making those corrects! Everything is working as intended now with both our RSpec file and the spec harness! 🎉

**Pull request merged**